src/perform_step/sdirk_perform_step.jl

```diff
@@ -512,6 +514,138 @@ end
   integrator.destats.nf += 1
   f(integrator.fsallast,u,p,t)
 end
+#like GenericImplicitEuler call twice with the intermediate calculation
+@muladd function perform_step!(integrator, cache::SDIRK22ConstantCache, repeat_step=false)
+  @unpack t,dt,uprev,u,f,p = integrator
+  @unpack uhold,rhs,nl_rhs = cache
+  alg = typeof(integrator.alg) <: CompositeAlgorithm ? integrator.alg.algs[integrator.cache.current] : integrator.alg
+  rhs.tmp = uprev
+  a,α,β = cache.tab
+  h̄ = a*dt
+
+  if integrator.success_iter > 0 && !integrator.reeval_fsal && alg.extrapolant == :interpolant
+    uhold[1] = current_extrapolant(t+h̄,integrator)
+  elseif alg.extrapolant == :linear
+    uhold[1] = uprev + h̄*integrator.fsalfirst
+  else # :constant
+    uhold[1] = uprev
+  end
+
+  rhs.t = t
+  rhs.dt = h̄
+  rhs.a = h̄
+  nlres = alg.nlsolve(nl_rhs,uhold)
+  uhold[1] = nlres[1]
+  integrator.fsallast = f(uhold[1],p,t+h̄)
+  ūₙ = uhold[1]
+
+  # intermediate
+  ūₙ₋₁ = α*uprev + β*ūₙ
+
+  if integrator.success_iter > 0 && !integrator.reeval_fsal && alg.extrapolant == :interpolant
+    uhold[1] = current_extrapolant(t+dt,integrator)
+  elseif alg.extrapolant == :linear
+    uhold[1] = ūₙ₋₁ + dt*integrator.fsalfirst
+  else # :constant
+    uhold[1] = ūₙ₋₁
+  end
+
+  rhs.t = t
+  rhs.dt = dt
+  rhs.a = dt
+  nlres = alg.nlsolve(nl_rhs,uhold)
+  uhold[1] = nlres[1]
+  integrator.fsallast = f(uhold[1],p,t+dt)
+  u = uhold[1]
+
+  if integrator.opts.adaptive && integrator.success_iter > 0
+    # local truncation error (LTE) bound by dt^2/2*max|y''(t)|
```

Comment:
does this error estimator make sense for this method?

Reply:
I think the method can be seen as two consecutive implicit Euler calculations, so I looked at how it was done for ```GenericImplicitEuler``` here:

https://github.com/JuliaDiffEq/OrdinaryDiffEq.jl/blob/abaf387213e1b1b020b795242cf4bd5ca2a950d8/src/perform_step/generic_implicit_perform_step.jl#L68

Is there a better way to do this?

Reply:
Is that the full leading truncation error term?

Reply:
I see, the method is 2nd order so the leading error term is

```julia
1/12 dt^3 max|y'''(t)|
```

I will update to reflect this.

Reply:
Take a look at the error estimator of the Trapezoidal method for inspiration. It's a little unstable to estimate the third derivative like that, but without a better error estimate that's about the best you can do.